src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,229 @@
+import React, {
+  Children,
+  ReactElement,
+  useEffect,
+  useRef,
+  useState,
+} from 'react'
+import classnames from 'classnames'
+interface TooltipProps {
+  label: string
+  position?: 'top' | 'bottom' | 'left' | 'right'
+  className?: string
+  children: ReactElement
+}
+
+const TRIANGLE_SIZE = 5
+const SPACER = 2
+const tooltipID = `tooltip-${Math.floor(Math.random() * 900000) + 100000}`
+
+function isElementInViewport(
```

Comment:
For some polish, I think this method is a good candidate to add to a `utils`-esque place. Suz has some good examples of that with DatePicker and the new [FileInput](https://github.com/trussworks/react-uswds/pull/900/files#diff-fed0c73187c20beede2d59813f693438b77907b5bbbfc1a54b3d59df4b0a6cfb)